src/PointwiseFunctions/AnalyticSolutions/NewtonianEuler/RiemannSolver.cpp

```diff
@@ -0,0 +1,534 @@
+  // (shock or rarefaction) on each side of the contact discontinuity
+  // must be sorted out. To do so, the first step is to compute the
+  // state variables in the "star" region by solving a transcendental equation,
+  // which is what all the math in this constructor is about.
+
+  // The initial guess to obtain the pressure in the star region is given by the
+  // Two-Shock approximation (Eqn. (4.47) in Toro), which gives best results
+  // overall. Other options are also possible (see Section 4.3.2 of Toro.)
+  const double guess_for_pressure_star = [&delta_u, this ](
+      const InitialData& left, const InitialData& right) noexcept {
+    // Eqn. (4.47) of Toro: guess derived from a linearized solution
+    // based on primitive variables.
+    double p_pv =
+        0.5 * (left.pressure_ + right.pressure_ -
+               0.25 * delta_u * (left.mass_density_ + right.mass_density_) *
+                   (left.sound_speed_ + right.sound_speed_));
+    p_pv = std::max(pressure_star_tol_, p_pv);
+
+    // Eqns. (4.48) of Toro: Two-Shock wave approximation.
+    const double g_left = sqrt(left.constant_a_ / (p_pv + left.constant_b_));
+    const double g_right = sqrt(right.constant_a_ / (p_pv + right.constant_b_));
+    return std::max(pressure_star_tol_, (g_left * left.pressure_ +
+                                         g_right * right.pressure_ - delta_u) /
+                                            (g_left + g_right));
+  }
+  (left_initial_data_, right_initial_data_);
+
+  // Compute bracket for root finder according to value of the function whose
+  // root we want (Eqn. 4.39 of Toro.)
+  FunctionOfPressureAndData<Dim> f_of_p_left(left_initial_data_,
+                                             adiabatic_index_);
+  FunctionOfPressureAndData<Dim> f_of_p_right(right_initial_data_,
+                                              adiabatic_index_);
+  const auto p_minmax = std::minmax(
+      {left_initial_data_.pressure_, right_initial_data_.pressure_});
+  const double f_min =
+      f_of_p_left(p_minmax.first) + f_of_p_right(p_minmax.first) + delta_u;
+  const double f_max =
+      f_of_p_left(p_minmax.second) + f_of_p_right(p_minmax.second) + delta_u;
+
+  double lower;
+  double upper;
+  if (f_min > 0.0 and f_max > 0.0) {
+    lower = 0.0;
+    upper = p_minmax.first;
+  } else if (f_min < 0.0 and f_max > 0.0) {
+    lower = p_minmax.first;
+    upper = p_minmax.second;
+  } else {
+    lower = p_minmax.second;
+    upper = 10.0 * lower;  // Arbitrary upper bound < \infty
+  }
+
+  // Now get pressure by solving transcendental equation. Newton-Raphson is OK.
+  const auto f_of_p_and_deriv =
+      [&f_of_p_left, &f_of_p_right, &delta_u ](const double pressure) noexcept {
+    // Function of pressure in Eqn. (4.5) of Toro.
+    return std::make_pair(
+        f_of_p_left(pressure) + f_of_p_right(pressure) + delta_u,
+        f_of_p_left.deriv(pressure) + f_of_p_right.deriv(pressure));
+  };
+  pressure_star_ =
```

Comment:
The root finder can throw an exception that should be caught and handle. For example, if I give values:
```
AnalyticSolution:
  RiemannSolver:
    AdiabaticIndex: 1.4
    InitialPosition: 0.25
    LeftMassDensity: 10.0
    LeftVelocity: [0.0]
    LeftPressure: 100.0
    RightMassDensity: 1.0
    RightVelocity: [0.0]
    RightPressure: 1.0
```
the solver fails but I just get a termination because of an uncaught exception.